packages/expo-sms/ios/ExpoSMSModule.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  ExpoSMSModule.swift
+//  ExpoSMS
+//
+//  Created by Alan Hughes on 11/11/2022.
+//
+
+import Foundation
+import ExpoModulesCore
+import WebKit
+import MessageUI
+import CoreServices
+import MobileCoreServices
+import UniformTypeIdentifiers
+
+struct ExpoSMSContext {
+    let promise: Promise
+    let smsDelgate: SMSDelegate
+}
+
+public class ExpoSMSModule: Module, SMSResultHandler {
+    private var smsContext: ExpoSMSContext?
+    private lazy var utils = appContext?.utilities
+
+    public func definition() -> ModuleDefinition {
+        Name("ExpoSMS")
+
+        AsyncFunction("isAvailableAsync") { (promise: Promise) in
+            promise.resolve(MFMessageComposeViewController.canSendText())
+        }
+
+        AsyncFunction("sendSMSAsync") {(addresses: [String], message: String, options: SmsOptions?, promise: Promise) in
+            sendSMSAsync(addresses: addresses, message: message, options: options, promise: promise)
+        }.runOnQueue(.main)
+    }
+
+    private func sendSMSAsync(addresses: [String], message: String, options: SmsOptions?, promise: Promise) {
+        if !MFMessageComposeViewController.canSendText() {
+            promise.reject("E_SMS_UNAVAILABLE", "SMS service not available")
+            return
+        }
+
+        if smsContext != nil {
+            promise.reject("E_SMS_SENDING_IN_PROGRESS",
+                           "Different SMS sending in progress. Await the old request and then try again.")
+            return
```

Comment:
Same here, throw an exception instead 😉 
Additionally, could you try to fit the reason message in one sentence and without the period at the end? We're trying to standardize that in the repo (I should put that to the contributing guide 😅)